src/sonic-framework/Makefile.am

```diff
@@ -5,9 +5,12 @@ BUILT_SOURCES = rebootbackend_protobuf_compilation rebootbackend_dbus_compilatio
 
 compile_protobufs: rebootbackend_protobuf_compilation rebootbackend_dbus_compilation
 
-rebootbackend_protobuf_compilation: 
-	/usr/bin/protoc --cpp_out=rebootbackend \
-		--proto_path=gnoi \
+update: 
+	mkdir -p build/gen
+
+rebootbackend_protobuf_compilation: update
```

Comment:
No need to make this a separate target. It is safe to call `mkdir -p` on an existing directory, and there is virtually no overhead.

```cpp
rebootbackend_protobuf_compilation: update
	mkdir -p build/gen
	/usr/bin/protoc --cpp_out=build/gen \
```

I would just combine it with the protoc rule.

If you want to keep it separate, please use a more descriptive name (e.g. "make_proto_build_dir").

Reply:
Addressed the comment.